### PR TITLE
Fix wrong cell_count and dgms list length

### LINKED ISSUE #33: Incorrect outputs of the flagser bindings.
#### Description
- The min and max dimensions parameter do not allow a barcode of the adequate dimension to be returned
- cell_count is only being returned if min_dim = 0 and max_dim = np.inf

The problem arises in C++ flagser and I have made a [PR](https://github.com/luetge/flagser/pull/23#issue-413544741) to flagser fixing it.

### PR DESCRIPTION
**Pull request by gtauzin**
#### Reference Issues/PRs
Fixes #33.

#### What does this implement/fix? Explain your changes.
- if min_dimension was 1, the 0-bars would always be outputed
- cell_count was only being outputed if min_dimension = 0 and max_dimension = inf
- Extends test to cell_count

### CODE REVIEW

File: pyflagser/flagio.py [hunk 1]
```diff
@@ -26,13 +26,13 @@ def loadflag(fname, fmt='csr', dtype=None):
...
         Matrix representation of a directed/undirected weighted/unweighted
         graph. Diagonal elements are vertex weights. The way zero values are
-        handled depends on the format of the matrix. If the matrix is a dense
-        `np.ndarray`, all zeros are explicitly accounted for and denote
-        zero-weight edges, i.e., edges appeaing at filtration value zero.
-        If the matrix is a sparse `scipy.sparse` matrix, zeros on the diagonal
-        and off-diagonal zeros assigned directly are treated explicitly. Off-
-        diagonal zeros that have not been assigned directly are treated
-        implicitly, i.e., corresponds to an abscence of edge.
+        handled depends on `fmt`. If `fmt` is ``'dense'``, all zeros are
```
Comment (ulupo) [inline pyflagser/flagio.py:29]: The diff here is a bit weird given that the current state of master agrees with what is proposed here: https://github.com/giotto-ai/pyflagser/blob/d59d35ecfc281be806535f647166775a0dcc215e/pyflagser/flagio.py#L29

@gtauzin, could you maybe pull from upstream master?

Comment (ulupo) [inline pyflagser/flagio.py:29]: The diff here is a bit weird given that the current state of master agrees with what is proposed here:

https://github.com/giotto-ai/pyflagser/blob/d59d35ecfc281be806535f647166775a0dcc215e/pyflagser/flagio.py#L29

@gtauzin, could you maybe pull from upstream master?

Comment (gtauzin) [reply pyflagser/flagio.py]: Yes, I am confused about the diff as well. I pulled from upstream master but the commit is empty.

Anyways, the code is right. It just includes commit that were made previously too.

Comment (ulupo) [reply pyflagser/flagio.py]: The diff looks much better now! Only 2 files changed, nothing that was already there. Thanks!

**Pull request merged**